### PR TITLE
Feature: Differentiate between cookies blocked in request or response

### LINKED ISSUE #381: Distinguish between cookie blocking on network response vs. network requests
## Feature Request

**Description:**
A cookie can blocked by Chrome when it is being set via a server response, as is the case with 3P cookies if 3P cookies being blocked. In this case the cookie will not be available at all. A cookie can also be accepted by Chrome, but later blocked from being set via certain requests. For example when having a 1P cookie with SameSite=Strict being sent to a subdomain of the top domain. 

**User Story:**
As a developer I need to distinguish scenarios where Chrome accepts a cookie on a network response (e.g. `Set Cookie` header), but that cookie is then blocked on some or all network requests (e.g. `Cookie` header). 

**Acceptance Criteria:**
- [x] Developer can distinguish both scenarios
- [x] Information about the number of requests for which a given cookie was blocked
- [x] Evaluate adding "Blocking direction" column, or keep that information on a per-cookie basis. 
- [ ] Filter cookie table by "Blocking direction"

### PR DESCRIPTION
**Pull request by ayushnirwal**
## Description

A cookie can be blocked in 2 ways - 
- Inbound - While accepting a cookie, a browser may block ( reject ) a cookie in the 'set-cookie' header.
- outbound - While associating a cookie to a request, a browser may decide not to add the cookie to the request's 'cookie' header.

This PR aims to handle that distinction while handling cookie-related information and displaying the data in the devtools panel.

An entry related to a cookie will be highlighted in yellow in the following cases.
- A cookie blocked to be saved in the browser in **all** of the `set-cookie` response headers.

An entry related to a cookie will have a warning symbol in front of the cell containing its name.
- A cookie blocked to be saved in the browser in **at least one but not all** of the `set-cookie` response headers.
- A cookie blocked to be associated with **at least one** of the requests.

## Relevant Technical Choices

- Add network event history array to the `CookieData` type.
- Add blocking status flags to the `CookieData` type.
- Add utility function `deriveBlockingStatus` to derive blocking status from network events history.
- Add enums for identifying network events while debugging and calculating `blocking status`.
- Update display components `bodyRow` and `bodyCell` to handle adding a warning icon before the cell's content.
- Update logic for highlighting a row in the component `bodyRow`.

## Screenshot/Screencast

Screenshot 2024-02-05 at 1 19 13 PM

---

## Checklist

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- [x] This code is covered by unit tests to verify that it works as intended.
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

Fixes #381

### CODE REVIEW

File: packages/extension/src/store/utils/deriveBlockingStatus.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    return false; // cookie is not blocked.
+  } else if (0 < numBlocked && numBlocked < CDPEvents.length) {
+    return null; // cookie may be blocked.
+  } else {
+    return true; // cookie is blocked.
+  }
+}
+
+/**
+ *
```
Comment (mohdsayed) [inline packages/extension/src/store/utils/deriveBlockingStatus.ts:68]: Please add missing descriptions.

File: tailwind.config.cjs [hunk 2]
```diff
@@ -127,6 +127,7 @@ module.exports = {
       'comet-grey': '#474747',
       'jet-black': '#202142',
       'warning-red': '#C33300',
+      'warning-orange': '#FE8d59',
```
Comment (mohdsayed) [inline tailwind.config.cjs:130]: This color is taken from DevTool right?

Comment (ayushnirwal) [reply tailwind.config.cjs:130]: Yes.

**Review by mohdsayed — APPROVED** [review]

**Review by amovar18 — APPROVED** [review]
LGTM!

**Pull request merged**